concepts/0346-didcomm-between-two-mobile-agents/README.md

```diff
@@ -0,0 +1,123 @@
+    "responseEndpoint": "https://cloudagenta.com/response", 
+    "consumer": "b1004443feff4f3cba25c45ef35b492c",
+    "consumerEndpoint" : "https://cloudagenta.com/consume"​
+}​
+```
+
+The registration data is base64 encrypted and is added to alink as part of the c_a_r query param. The recipient key is the public key of "Cloud Agent A". The service endpoint is where the edge agent should send the message to. Response endpoint is where a response that is being sent to Alice should be sent to. For example, if Bob wants to send a message to Alice, then Bob should send the message to the response endpoint. Consumer endpoint is where Alice's edge agent should consume the messages that are sent to her. The "Consumer" is an identifier to identify Alice's edge agent by the cloud agent "A". This identifier is different with each cloud agent and hence provides low correlation risk. Each time an invitation QR code is generated, a new consumer id is generated. No acknowledgment is required to be sent to the cloud agent or vice versa as the consumer-generated is never repeated.
+
+All the endpoint data and the public key of the cloud agents are then stored as non secret records in Alice's wallet with a tag "cloud-agent"
+
+## How connection request from Alice flows to Bob
+
+When Alice scans Bob's QR code invitation. It starts preparing the connection request message. It first queries the wallet record service for records tagged with "cloud-agent" and puts them in a list. The edge agent now randomly chooses one from the list (say Cloud Agent "A") and creates a new list without the cloud agent that is already chosen. Alice's edge agent creates the connection request message json and adds the service endpoint as the chosen cloud agent's response endpoint together with its consumer id. 
+
+```JSON
+"serviceEndpoint": "https://cloudagenta.com/response/b1004443feff4f3cba25c45ef35b492c"
+```
+
+It then packs this message by Bob's recipient key and then creates another json message structure like the below
+
+```JSON
+{​
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/cloudagentmessageforward",​
+    "@id": "12345678900987654321",​
+    "label": "CloudAgentA",​
+    "message": "<Encrypted message for Bob here>",
+    "forwardTo": "<Service endpoint of Bob>"​
+}​
+```
+
+It then packs it with the public key of cloud agent "A".
+
+Now it randomly chooses cloud agent from the new list and keeps on repeating the process of writing the message forwarding request.
+
+For example, say the next random cloud agent that it chooses is Cloud Agent "C". So now it creates another message forward json structure as below
+
+```JSON
+{​
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/cloudagentmessageforward",​
+    "@id": "12345678900987654321",​
+    "label": "CloudAgentC",​
+    "message": "<Encrypted message for Cloud Agent A>",
+    "forwardTo": "<Service endpoint of Cloud Agent A>"​
```

Comment:
How is this special type of forward message different from a generic `forward` message (which is regularly used to forward from cloud agents to mobile agents already)? Can the two be reconciled?

Reply:
@dhh1128 You are right I forgot to change this part. This was my initial idea when I did not know about the Forward Message Type, but now in my working code I am actually using the same. I wil change it here too

Reply:
@dhh1128 have changed now.Please have a look